src/vuln_analysis/tools/tests/test_transitive_code_search.py

```diff
@@ -120,3 +124,110 @@ async def get_transitive_code_runner_function():
     transitive_code_search = transitive_search(config=TransitiveCodeSearchToolConfig(), builder=None)
     async for function in transitive_code_search.gen:
         return function.single_fn
+
+python_dependency_tree_mock_output = (
+    'deptree==0.0.12  # deptree\n'
+    '  importlib-metadata==8.7.0  # importlib-metadata\n'
+    '    zipp==3.23.0  # zipp>=3.20\n'
+    '  setuptools==80.9.0  # setuptools\n'
+    'flask==3.1.2  # flask\n'
+    '  blinker==1.9.0  # blinker>=1.9.0\n'
+    '  click==8.3.0  # click>=8.1.3\n'
+    '  itsdangerous==2.2.0  # itsdangerous>=2.2.0\n'
+    '  jinja2==3.1.6  # jinja2>=3.1.2\n'
+    '    markupsafe==3.0.3  # MarkupSafe>=2.0\n'
+    '  markupsafe==3.0.3  # markupsafe>=2.1.1\n'
+    '  werkzeug==3.1.3  # werkzeug>=3.1.0\n'
+    '    markupsafe==3.0.3  # MarkupSafe>=2.1.1\n'
+    '    mock-package==1.1.1'
+)
+
+def mock_file_open(*args, **kwargs):
+    file_path = args[0] if args else kwargs.get('file', '')
+    mock_file = MagicMock()
+    mock_file.__enter__ = MagicMock(return_value=mock_file)
+    mock_file.__exit__ = MagicMock(return_value=None)
+    if 'ecosystem_data.txt' in str(file_path):
+        mock_file.read.return_value = "PYTHON"
+    elif 'requirements.txt' in str(file_path):
+        mock_file.__iter__ = MagicMock(return_value=iter([
+            "flask==2.0.1\n",
+            "werkzeug==2.0.1\n",
+            "requests==2.25.1\n"
+        ]))
+    else:
+        mock_file.read.return_value = ""
+    return mock_file
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("test_case", [
+    {
+        "name": "python_1",
+        "search_query": "werkzeug,formparser.MultiPartParser",
+        "expected_path_found": True,
+        "expected_list_length": 2,
+        "mock_documents": [python_script_example, python_init_function_example, python_full_document_example]
+    },
+    {
+        "name": "python_2",
+        "search_query": "werkzeug,formparser.MultiPartParser.parse",
+        "expected_path_found": False,
+        "expected_list_length": 0,
+        "mock_documents": [python_script_example, python_init_function_example, python_full_document_example, python_parse_function_example]
+    },
+    {
+        "name": "python_3",
+        "search_query": "mock_package,mock_function_in_use",
+        "expected_path_found": True,
+        "expected_list_length": 3,
+        "mock_documents": [python_script_example, python_init_function_example, python_full_document_example, python_parse_function_example, python_mock_function_in_use, python_mock_file]
+    }
+])
+@patch('vuln_analysis.utils.dep_tree.run_command', return_value=python_dependency_tree_mock_output)
+@patch('builtins.open', side_effect=mock_file_open)
+async def test_transitive_search_parameterized(mock_open, mock_run_command,test_case):
```

Comment:
@TamarW0 Small adjustment
```suggestion
async def test_transitive_search_python_parameterized(mock_open, mock_run_command,test_case):
```